client/src/extension.ts

```diff
@@ -47,61 +53,103 @@ export function activate(context: ExtensionContext) {
         documentSelector: [{ scheme: 'file', language: 'yml' }],
         synchronize: {
             // Synchronize the setting section 'yseopml' to the server
-            configurationSection: 'yseopml',
+            configurationSection: yseopmlSectionName,
             // Notify the server about file changes to '.clientrc files contain in the workspace
             fileEvents: workspace.createFileSystemWatcher('**/.clientrc'),
         },
     };
 
-    yseopCliPath = workspace.getConfiguration('yseopml').get('pathToYseopCli');
+    const yseopmlConfig: WorkspaceConfiguration = workspace.getConfiguration(yseopmlSectionName);
+    yseopCliPath = yseopmlConfig.get(pathToYseopCliKey);
+    parseAllProjectFilesAtStartup = yseopmlConfig.get(parseWholeProjectKey);
 
-    workspace.onDidChangeConfiguration((event) => {
-        if (event.affectsConfiguration('yseopml.pathToYseopCli')) {
-            yseopCliPath = workspace.getConfiguration('yseopml').get('pathToYseopCli');
-        }
+    workspace.onDidChangeConfiguration(() => {
+        yseopCliPath = yseopmlConfig.get(pathToYseopCliKey);
+        parseAllProjectFilesAtStartup = yseopmlConfig.get(parseWholeProjectKey);
```

Comment:
The lines

```
yseopCliPath = yseopmlConfig.get(pathToYseopCliKey);
parseAllProjectFilesAtStartup = yseopmlConfig.get(parseWholeProjectKey);
```

appear twice. Consider making a `readConfig(yseopmlConfig)` function or something.

Btw, are we sure the “DidChangeConfiguration” isn’t triggered at startup? That would make the instructions above redundant. But I think we already investigated something like this.